bin/check-ssl-hsts-status.rb

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+# encoding: UTF-8
+#  check-ssl-hsts-preload.rb
+#
+# DESCRIPTION:
+#   Checks a domain against the chromium HSTS API reporting on the preload status of the domain
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#   # Basic usage
+#   check-ssl-hsts-preload.rb -d <domain_name>
+#   # Specify the CRITICAL and WARNING alerts to either unknown (not in the database), pending or preloaded
+#   check-ssl-hsts-preload.rb -d <domain_name> -c <critical_alert> -w <warning_alert>
+#
+# LICENSE:
+#   Copyright 2017 Rowan Wookey <admin@rwky.net>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE for
+#   details.
+#   
+#   Inspired by https://github.com/sensu-plugins/sensu-plugins-ssl/blob/master/bin/check-ssl-qualys.rb Copyright 2015 William Cooke <will@bruisyard.eu>
+#
+
+require 'sensu-plugin/check/cli'
+require 'json'
+require 'net/http'
+
+class CheckSSLHSTSPreload < Sensu::Plugin::Check::CLI
+
+  STATUSES = ['unknown', 'pending', 'preloaded'].freeze
+  API_URL = 'https://hstspreload.org/api/v2/status'.freeze
+
+  option :domain,
+         description: 'The domain to run the test against',
+         short: '-d DOMAIN',
+         long: '--domain DOMAIN',
+         required: true
+
+  option :warn,
+         short: '-w STATUS',
+         long: '--warn STATUS',
+         description: 'WARNING if this status or worse',
+         in: STATUSES,
+         default: 'pending'
+
+  option :critical,
+         short: '-c STATUS',
+         long: '--critical STATUS',
+         description: 'CRITICAL if this status or worse',
+         in: STATUSES,
+         default: 'unknown'
+
+  def run
+    uri       = URI(API_URL)
+    uri.query = URI.encode_www_form({ domain: config[:domain] })
+    response  = Net::HTTP.get_response(uri)
+    warning 'Bad response recieved from API' unless response.is_a?(Net::HTTPSuccess)
```

Comment:
I think we might want to handle a redirection rather than simply saying its bad.